File: codility/lesson05-prefix-sums/PassingCars.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)
    vector<int> cum_num_west(A);
    long long ans = 0;
    for(int i = A.size()-2; i >= 0; --i){
        cum_num_west[i] += cum_num_west[i+1];
        if(A[i] == 0) ans += cum_num_west[i];
    }
    if(ans > 1e+9) return -1;
    else return ans;
}

File: codility/PassingCars.py
def solution(A):
    # write your code in Python 3.6
    N = len(A)

    ans = 0
    cum_sum = A.copy()
    for i in reversed(range(N-1)):
        cum_sum[i] += cum_sum[i+1]
        if A[i] == 0:
            ans += cum_sum[i]
    
    if ans > int(1e+9):
        return -1
    return ans